### PR TITLE
Disallow grants for broadcast channels

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

We cannot allow the same grant to be reused across different messages

## Proposal

Disallow that with an error for now

## Test Plan

CI

### CODE REVIEW

**Comment by andresilva91** [comment]
* **#1941** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1941?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by andresilva91** [comment]
I just (maybe a bit lazily) modified `run_test_handle_certificates_to_create_application` to include a failing broadcast message. Lmk if you guys think I should create a whole new test for this.

File: linera-chain/src/lib.rs [hunk 1]
```diff
@@ -136,6 +136,8 @@ pub enum ChainError {
     ClosedChain,
     #[error("All operations on this chain must be from one of the following applications: {0:?}")]
     AuthorizedApplications(Vec<ApplicationId>),
+    #[error("Can't reuse grant across different broadcast messages")]
+    GrantReuseOnBroadcast,
```
Comment (ma2bd) [inline linera-chain/src/lib.rs:140]: `Reuse` -> `Use`

**Review by ma2bd — APPROVED** [review]
Thanks @andresilva91 !

File: linera-core/src/unit_tests/wasm_worker_tests.rs [hunk 2]
```diff
@@ -201,16 +201,40 @@ where
...
+        block: broadcast_block.clone(),
+        messages: vec![failing_broadcast_outgoing_message],
+        message_counts: vec![1],
+        state_hash: publisher_state_hash,
+    });
+    let failing_broadcast_certificate =
+        make_certificate(&committee, &worker, failing_broadcast_block_proposal);
+
+    worker
+        .fully_handle_certificate(failing_broadcast_certificate.clone(), vec![])
```
Comment (afck) [inline linera-core/src/unit_tests/wasm_worker_tests.rs:223]: No need to `clone()`.

**Pull request merged**